mmv1/third_party/terraform/services/resourcemanager/resource_google_project_iam_member_remove.go

```diff
@@ -0,0 +1,132 @@
+				ForceNew:    true,
+				Required:    true,
+				Description: `The target role that should be removed.`,
+			},
+			"member": {
+				Type:        schema.TypeString,
+				ForceNew:    true,
+				Required:    true,
+				Description: `The IAM principal that should not have the target role.`,
+			},
+		},
+		UseJSONNumber: true,
+	}
+}
+
+func resourceGoogleProjectIamMemberRemoveCreate(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*transport_tpg.Config)
+
+	project := d.Get("project").(string)
+	role := d.Get("role").(string)
+	member := d.Get("member").(string)
+
+	found := false
+	iamPolicy, err := config.NewResourceManagerClient(config.UserAgent).Projects.GetIamPolicy(project,
+		&cloudresourcemanager.GetIamPolicyRequest{
+			Options: &cloudresourcemanager.GetPolicyOptions{
+				RequestedPolicyVersion: tpgiamresource.IamPolicyVersion,
+			},
+		}).Do()
+	if err != nil {
+		return transport_tpg.HandleNotFoundError(err, d, d.Id())
+	}
+
+	for i := 0; i < len(iamPolicy.Bindings); i++ {
+		if role == iamPolicy.Bindings[i].Role {
+			for j := 0; j < len(iamPolicy.Bindings[i].Members); j++ {
+				if member == iamPolicy.Bindings[i].Members[j] {
+					found = true
+					iamPolicy.Bindings[i].Members = append(iamPolicy.Bindings[i].Members[:j], iamPolicy.Bindings[i].Members[j+1:]...)
+					break
+				}
+			}
+		}
+	}
+
+	if found == false {
+		fmt.Printf("[DEBUG] Could not find Member %s with the corresponding role %s. No removal necessary", member, role)
+	} else {
+		updateRequest := &cloudresourcemanager.SetIamPolicyRequest{
+			Policy:     iamPolicy,
+			UpdateMask: "bindings",
+		}
+		_, err = config.NewResourceManagerClient(config.UserAgent).Projects.SetIamPolicy(project, updateRequest).Do()
+		if err != nil {
+			return fmt.Errorf("cannot update IAM policy on project %s: %v", project, err)
+		}
+	}
+
+	d.SetId(fmt.Sprintf("%s/%s/%s", project, member, role))
+
+	return resourceGoogleProjectIamMemberRemoveRead(d, meta)
+}
+
+func resourceGoogleProjectIamMemberRemoveRead(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*transport_tpg.Config)
+
+	project := d.Get("project").(string)
+	role := d.Get("role").(string)
+	member := d.Get("member").(string)
+
+	found := false
+	iamPolicy, err := config.NewResourceManagerClient(config.UserAgent).Projects.GetIamPolicy(project,
+		&cloudresourcemanager.GetIamPolicyRequest{
+			Options: &cloudresourcemanager.GetPolicyOptions{
+				RequestedPolicyVersion: tpgiamresource.IamPolicyVersion,
+			},
+		}).Do()
+	if err != nil {
+		return transport_tpg.HandleNotFoundError(err, d, d.Id())
+	}
+
+	for i := 0; i < len(iamPolicy.Bindings); i++ {
+		if role == iamPolicy.Bindings[i].Role {
+			for j := 0; j < len(iamPolicy.Bindings[i].Members); j++ {
+				if member == iamPolicy.Bindings[i].Members[j] {
+					found = true
+					break
+				}
+			}
+		}
+	}
+
+	if found {
+		fmt.Printf("[DEBUG] found policy for resource %v, removing from state", d.Id())
+		d.SetId("")
```

Comment:
So if the IAM member exists we consider this resource not present to trigger the recreation that removes the IAM member? I think that makes sense

Reply:
Yep! From docs:

> If the membership is ever re-added, the next refresh will clear this
resource from state, proposing re-adding it to correct the membership.

Reply:
I considered changing it to an output boolean we change in a CustomizeDiff, which would make adding import make sense, but figured it wasn't worth the extra effort on the initial implementation. We could always change that in a major version (and potentially even a minor depending how strict we are 🤷‍♀️ )